pkg/api/resource/quantity.go

```diff
@@ -362,11 +367,35 @@ func (q *Quantity) Sub(y Quantity) error {
 		q.Amount = &inf.Dec{}
 		return q.Sub(y)
 	default:
+		// we want to preserve the format of the non-zero value
+		zero := &inf.Dec{}
+		if q.Amount.Cmp(zero) == 0 && y.Amount.Cmp(zero) != 0 {
+			q.Format = y.Format
+		}
 		q.Amount.Sub(q.Amount, y.Amount)
 	}
 	return nil
 }
 
+// Neg sets q to the negative value of y.
+// It updates the format of q to match y.
+func (q *Quantity) Neg(y Quantity) error {
+	switch {
+	case y.Amount == nil:
+		q.Amount = &inf.Dec{}
+		if len(y.Format) != 0 {
```

Comment:
Shouldn't q just get totally obliterated no matter what? It's not uncommon to chain function calls such as this where `q` is just some temporary storage that you're going to reuse over again. So `q` should be pure output, we should not let any part of it survive.